BlobStorage/Advanced.cs

```diff
@@ -1678,17 +1413,17 @@ private static async Task CopyLargeBlockBlobAsync(CloudBlobContainer container,
         /// <summary>
         /// Uploads the blob as a set of 256 KB blocks.
         /// </summary>
-        /// <param name="container">A CloudBlobContainer object.</param>
+        /// <param name="container">A BlobContainerClient object.</param>
         /// <returns>A Task object.</returns>
-        private static async Task UploadBlobInBlocksAsync(CloudBlobContainer container)
+        private static async Task UploadBlobInBlocksAsync(BlobContainerClient container)
```

Comment:
This method can be rewritten to be less than half the size. Instead of doing all this manual copying from one buffer to another, `MemoryStream` has a really useful constructor: `MemoryStream(byte[] buffer, int index, int count)`. We can create `MemoryStream`s over the existing buffer instead of all this copying, which will also reduce memory overhead.

We can replace nearly all this with a loop: `for (int index = 0; offset < randomBytes.Length; index += blockSize)` and then just construct `MemoryStream`s for upload as `new MemoryStream(randomBytes, index, Math.Min(blockSize, randomBytes.Length - offset)`.